src/Microsoft.AspNetCore.Authorization/DefaultAuthorizationService.cs

```diff
@@ -51,16 +52,38 @@ public DefaultAuthorizationService(IAuthorizationPolicyProvider policyProvider,
 
             if (authContext.HasSucceeded)
             {
-                _logger.UserAuthorizationSucceeded(user?.Identity?.Name);
+                _logger.UserAuthorizationSucceeded(GetUserNameForLogging(user));
                 return true;
             }
             else
             {
-                _logger.UserAuthorizationFailed(user?.Identity?.Name);
+                _logger.UserAuthorizationFailed(GetUserNameForLogging(user));
                 return false;
             }
         }
 
+        private string GetUserNameForLogging(ClaimsPrincipal user)
+        {
+            var identity = user?.Identity;
+            if (identity != null)
+            {
+                var name = identity.Name;
+                if (name != null)
+                {
+                    return name;
+                }
+                return GetClaimValue(identity, "sub")
```

Comment:
Well we are checking Identity.Name first, but the order comes from @blowdart, I'll let you guys hash out the appropriate priority :)